libraries/standard/http/src/http_client.c

```diff
@@ -929,16 +1031,210 @@ HTTPStatus_t HTTPClient_Send( const HTTPTransportInterface_t * pTransport,
+                             fieldLen, pContext->pField );
+
+        /* Set the flag to indicate that header has been found in response. */
+        pContext->fieldFound = 1u;
+    }
+    else
+    {
+        /* Empty else for MISRA 15.7 compliance. */
+    }
+
+    return HTTP_PARSER_CONTINUE_PARSING;
+}
+
+/*-----------------------------------------------------------*/
+
+static int findHeaderValueParsedCallback( http_parser * pHttpParser,
+                                          const char * pVaLueLoc,
+                                          size_t valueLen )
+{
+    findHeaderContext_t * pContext = ( findHeaderContext_t * ) pHttpParser->data;
+    int retCode = HTTP_PARSER_CONTINUE_PARSING;
+
+    assert( pHttpParser != NULL );
+    assert( pVaLueLoc != NULL );
+    assert( valueLen > 0u );
+
+    assert( pContext->pField != NULL );
+    assert( pContext->fieldLen > 0u );
+    assert( pContext->pValueLoc != NULL );
+    assert( pContext->pValueLen != NULL );
+
+    /* The header value found flag should not be set. */
+    assert( pContext->valueFound == 0u );
+
+    if( pContext->fieldFound == 1u )
+    {
+        IotLogDebugWithArgs( "Found header value in response: "
+                             "RequestedField=%.*s, ValueLocation=0x%d",
+                             pContext->fieldLen, pContext->pField, pVaLueLoc );
+
+        /* Populate the output parameters with the location of the header value in the response buffer. */
+        *pContext->pValueLoc = ( const uint8_t * ) pVaLueLoc;
+        *pContext->pValueLen = valueLen;
+
+        /* Set the header value found flag. */
+        pContext->valueFound = 1u;
+
+        /* As we have found the value associated with the header, we don't need
+         * to parse the response any further. */
+        retCode = HTTP_PARSER_STOP_PARSING;
+    }
+    else
+    {
+        /* Empty else for MISRA 15.7 compliance. */
+    }
+
+    return retCode;
+}
+
+/*-----------------------------------------------------------*/
+
+static int findHeaderOnHeaderCompleteCallback( http_parser * pHttpParser )
+{
+    /* Disable unused parameter warning. */
+    ( void ) pHttpParser;
+
+    assert( pHttpParser != NULL );
+
+    /* If we have reached here, all headers in the response have been parsed but the requested
+     * header has not been found in the response buffer. */
+    IotLogDebugWithArgs( "Reached end of header parsing: Requested header not found in response: "
+                         "HeaderField=%.*s",
+                         ( ( findHeaderContext_t * ) pHttpParser->data )->fieldLen,
+                         ( ( findHeaderContext_t * ) pHttpParser->data )->pField );
+
+    /* No further parsing is required; thus, indicate the parser to stop parsing. */
+    return HTTP_PARSER_STOP_PARSING;
+}
+
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t findHeaderInResponse( const uint8_t * pBuffer,
```

Comment:
Does findHeaderInResponse.c.gcov have 100% coverage?
```
function findHeaderInResponse called N returned 100% blocks executed 100%
```

Reply:
Nope, this is what I get 
`function findHeaderInResponse called 10 returned 100% blocks executed 90%`